harold/plugins/deploy.py

```diff
@@ -156,6 +173,28 @@ def _handle_request(self, request):
         self.monitor.onPushProgress(salon, id, host, index)
 
 
+class DeployHoldAllListener(DeployListener):
+    """
+    Trigger a deploy hold for all salons
+    """
+    isLeaf = True
+
+    def _handle_request(self, request):
+        reason = request.args['reason'][0]
+        type = request.args['type'][0]
+        self.monitor.hold_all(DeployHoldType.__members[type], reason)
```

Comment:
https://cpython-test-docs.readthedocs.io/en/latest/library/enum.html

Reply:
https://docs.python.org/3/library/enum.html#programmatic-access-to-enumeration-members-and-their-attributes

`DeployHoldType[type]` should be sufficient?

Reply:
```
Python 2.7.10 (default, Oct  6 2017, 22:29:07) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import Enum
>>> class DeployHoldType(Enum):
...     code_freeze = 'Code Freeze'
...     manual = 'Manual'
... 
>>> DeployHoldType['manual']
<DeployHoldType.manual: 'Manual'>
>>> DeployHoldType['code_freeze']
<DeployHoldType.code_freeze: 'Code Freeze'>
>>> 
```

Reply:
Yup

Reply:
Since I can't pass the `type` in I will just remove this and detect the type based on the reason string.